lib/apis/v3/node.go

```diff
@@ -98,6 +103,27 @@ type NodeWireguardSpec struct {
 	InterfaceIPv4Address string `json:"interfaceIPv4Address,omitempty" validate:"omitempty,ipv4"`
 }
 
+// NodeAddressType represents a node address type as a string
+type NodeAddressType string
+
+const (
+	NodeAddressBGPIP             NodeAddressType = "BGPIP"
+	NodeAddressVXLANTunnelIP     NodeAddressType = "VXLANTunnelIP"
+	NodeAddressVXLANMAC          NodeAddressType = "VXLANMAC"
+	NodeAddressIPIPTunnelIP      NodeAddressType = "IPIPTunnelIP"
+	NodeAddressWireguardTunnelIP NodeAddressType = "WireguardTunnelIP"
+	NodeAddressKubeInternalIP    NodeAddressType = "KubeInternalIP"
+	NodeAddressKubeExternalIP    NodeAddressType = "KubeExternalIP"
+)
+
+// NodeAddress represents an address assigned to the node annotated with its type.
+type NodeAddress struct {
+	// Type denotes what sort of address it is.
+	Type NodeAddressType `json:"type"`
+	// Address is a string rpresentation of the actual address.
+	Address string `json:"address"`
```

Comment:
We probably want to implement validation on this struct and its fields, right?

Reply:
Since we never use a value written by a user, it feels redundant.

Reply:
I thought the plan was to make this field writable for etcd mode to work. Right?

Reply:
Right, at some point, yes. I was pondering validation. For the type, if it is unknown, the consumer skips it, as it does not understand the address. If the consumer is older, there is a chance that it does not know a new but validated type. For the address itself, we would need to validate the combinations address-type. If we leave taking care of duplicities to the user, why not to let it check that an IP is an IP (unless we provide types for 4/6 it needs to figure that already) etc.

But I am happy to add a validator for the tupple.